portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject res = new JsonObject();
+
+    try {
+      int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
+
+      res.add("comments", commentsJson);
+      res.addProperty("success", true);
+      } catch (Exception e) {
+        res.addProperty("success", false);
+        String errorMessage = e.getMessage();
 
-    Gson gson = new Gson();
+        if (errorMessage == "null") {
+          errorMessage = "Please enter a value for maxComments.";
+        }
+        else {
+          errorMessage = "NumberFormatException: " + errorMessage;
```

Comment:
I think the first error message is a good one, but the second could be more clear - it reads like an internal error message rather than the kind of thing you'd tell to a potentially less-tech-savvy client.